File: algorithm/C++/0118-Pascal's_Triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> results = {{1}};
        for (int i=1; i<numRows; ++i) {
            vector<int> current_result;
            for (int j=0; j<=i; ++j) {
                if (j == 0) {
                    current_result.push_back(1);
                } else if (j == i) {
                    current_result.push_back(1);
                } else {
                    current_result.push_back(results[i-1][j-1]+results[i-1][j]);
                }
            }
            results.push_back(current_result);
        }
        return results;
    }
};

File: algorithm/Java/0118-Pascal's_Triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> triangle = new ArrayList<List<Integer>>();
        triangle.add(new ArrayList<Integer>());
        triangle.get(0).add(1);
        for (int i=1; i<numRows; ++i) {
            List<Integer> row = new ArrayList<>();
            List<Integer> prev = triangle.get(i - 1);
            for (int j=0; j<=i; ++j) {
                if (j == 0) {
                    row.add(1);
                } else if (j == i) {
                    row.add(1);
                } else {
                    row.add(prev.get(j) + prev.get(j - 1));
                }
            }
            triangle.add(row);
        }
        return triangle;
    }
}
// Time : O(n^2)
// Space : O(n^2)